python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+                                       self.config.readNoise / self.config.gain)
+
+                # 8. Apply serial overscan in ADU
+                # Apply gradient along X axis to the whole amp
+                # First the serial overscan and corners region
+                self.amplifierAddXGradient(serialOscanData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+                # And second the image and parallel overscan region
+                self.amplifierAddXGradient(ampData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+                self.amplifierAddXGradient(parallelOscanData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+
+        if self.config.doGenerateAmpDict:
+            expDict = dict()
+            for amp in exposure.getDetector():
+                expDict[amp.getName()] = exposure
+            return expDict
+        else:
+            return exposure
+
+    def addGain(self, ampData, gain):
+        ampArr = ampData.array
+        ampArr[:] = ampArr[:] / gain
+
+    def amplifierAddXGradient(self, ampData, start, end):
+        """Add a x-axis linear gradient to an amplifier's image data.
+
+         This method operates in the amplifier coordinate frame.
+
+        Parameters
+        ----------
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        start : `float`
+            Start value of the gradient (at y=0).
+        end : `float`
+            End value of the gradient (at y=ymax).
+        """
+        nPixX = ampData.getDimensions().getX()
+        ampArr = ampData.array
+        ampArr[:] = ampArr[:] + (np.interp(range(nPixX), (0, nPixX - 1), (start, end)).reshape(1, nPixX)
+                                 + np.zeros(ampData.getDimensions()).transpose())
+
+
+class RawMockLSST(IsrMockLSST):
+    """Generate a raw exposure suitable for ISR.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.config.isTrimmed = False
+        self.config.doGenerateImage = True
+        self.config.doGenerateAmpDict = False
+
+        # Add astro effects
+        self.config.doAddSky = True
+        self.config.doAddSource = True
+
+        # Add optical effects
+        self.config.doAddFringe = True
+
+        # Add instru effects
+        self.config.doAddParallelOverscan = True
+        self.config.doAddSerialOverscan = True
+        self.config.doAddCrosstalk = True
+        self.config.doAddBias = True
+        self.config.doAddDark = True
+
+        self.config.doAddFlat = True
+
+
+class CalibratedRawMockLSST(RawMockLSST):
+    """Generate a trimmed raw exposure.
+    """
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+#        self.config.isTrimmed = True
```

Comment:
If this is not trimmed, then the overscan regions should be filled with read noise.  I also think that `doAddFringe` should be `False`, as the calibration should have corrected that.  I'm happy to be wrong here.

Reply:
I would agree but the fringe correction is done after the ISR corrections so I am leaving them here. Happy to change if needed later on.